submariner/cable-driver-protocols-enhancement.md

```diff
@@ -0,0 +1,179 @@
+# Cable Driver Protocol Support Enhancements
+
+## Summary
+
+Submariner today supports three cable drivers: VXLAN, IPSec (tunnel mode) and WireGuard. The
+goal of this proposal is to outline additional protocols and enhancements for the Submariner
+Cable Driver. These include the additional support of IP-in-IP for unencrypted connections
+between clusters and the support of IPSec transport mode.
+
+For IP in IP, the benefits fall into the following categories:
+
+* **Performance**: IP in IP is has a lower overhead than VXLAN which could lead to performance
+improvements in terms of packet overhead (bandwidth consumption) as well as the cost of processing
+packets.
+* **Interoperability**:IP in IP can be used to transport packets between domains when the protocol
+in those domains (e.g. IPv6) is not supported by intermediary networks (e.g. IPv4).
+
+For Overlay-over-IPSec in transport mode with further configuration, the benefits fall into the
```

Comment:
I'm not familiar with SM IpSec implmenetation, 
could you please elaborate more about this option ? and how it differs from the current IpSec implmenetation ?

Reply:
OK, I tcpdumpped SM ipsec traffic.

So, you mean that for VxLAN over ipsec for example, the packet format will be something like: 

[L2]-[external-IP][UDP][VxLan Header] [org_L2][org_ip header][ESP of org IP payload]

While current SM ipsec packet format (tunnel mode)  is : 
[L2]-[external-IP] [ESP of org IP payload including org_ip header]

Could you please elaborate on the performance improvement you mentioned?  is it related to HW offload?

Reply:
Yeah - the performance improvements are mostly related to HW offload... 

reducing the packet overhead and thus the amount of cycles needed to process the packet also helps from a CPU and bw consumption POV... 

There's a nice illustration of Tunnel mode vs transport mode packets here https://yurmagccie.files.wordpress.com/2019/01/ipsec-tunnel-transport.jpg